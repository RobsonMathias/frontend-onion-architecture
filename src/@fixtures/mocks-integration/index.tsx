export * from './store.mock';
export * from './adapters.mock';
export * from './../mocks/http.mock';
export * from './../mocks/router.mock';
