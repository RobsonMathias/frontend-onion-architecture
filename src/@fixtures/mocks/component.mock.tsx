import React from 'react';

export function Mock() {
  return <div />;
}

export const mockComponent = (componentsName: string[] = []) => {
  let composed: { [key: string]: React.ReactNode } = {};
  componentsName.forEach((name: string) => {
    composed[name] = Mock;
  });
  return composed;
};
