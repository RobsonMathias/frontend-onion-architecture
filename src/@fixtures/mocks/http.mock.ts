export const mockHttp = {
  patch: jest.fn(),
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
  delete: jest.fn(),
  interceptors: {
    request: {
      use: jest.fn(),
      eject: jest.fn()
    },
    response: {
      use: jest.fn()
    }
  }
} as any;
