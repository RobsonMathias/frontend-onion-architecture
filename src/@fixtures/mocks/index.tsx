export * from './store.mock';
export * from './router.mock';
export * from './component.mock';
export * from './wrappers';
export * from './http.mock';
export * from './services.mock';
export * from './navigation.mock';
export * from './storage.mock';
export * from './use-auth.mock';
