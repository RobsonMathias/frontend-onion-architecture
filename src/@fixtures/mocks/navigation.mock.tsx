import React from 'react';

const setData = jest.fn();
const mockUseUI = jest.fn(() => ({ setData }));

const query = { get: jest.fn(), set: jest.fn() };
const set = jest.fn();
const setAll = jest.fn();
const mockUseQuery = jest.fn(() => ({ query, setAll, set }));

const replace = jest.fn();
const mockUseURL = jest.fn(() => ({ replace }));

const mockNavBar = React.forwardRef((props, ref) => <div {...props} />);

export const mockNavigation = {
  NavBar: mockNavBar,
  useQuery: mockUseQuery,
  query,
  setAll,
  set,
  useUI: mockUseUI,
  setData,
  useURL: mockUseURL,
  replace
};
