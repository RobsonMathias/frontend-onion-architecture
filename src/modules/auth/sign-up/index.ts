export * from './sign-up.container';
