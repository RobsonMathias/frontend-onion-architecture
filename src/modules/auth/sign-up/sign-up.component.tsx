import React, { useEffect } from 'react';

import { SignUpPayload } from '@types';
import { Box, Button, TextField } from '@mui/material';
import { FieldErrors, useForm } from 'react-hook-form';

type SignUpComponentProps = {
  onSubmit: (payload: SignUpPayload) => void;
  loading?: boolean;
};

export function SignUpComponent(props: SignUpComponentProps) {
  const { onSubmit, loading } = props;
  const {
    register,
    handleSubmit,
    reset,
    formState: { errors, isSubmitted }
  } = useForm({ mode: 'all' });

  useEffect(() => {
    return () => {
      reset();
    };
  }, []);

  function hasError(error?: FieldErrors): string {
    const { message } = error || { message: '' };
    return isSubmitted ? message : '';
  }

  function submit(data: SignUpPayload) {
    onSubmit(data);
  }

  return (
    <form onSubmit={handleSubmit(submit)}>
      <Box mb={2}>
        <TextField
          variant="standard"
          fullWidth={true}
          type="email"
          inputProps={{ 'data-testid': 'email' }}
          label="E-mail"
          error={!!hasError(errors.email)}
          helperText={hasError(errors.email)}
          placeholder="email@exemplo.com"
          {...register('email', {
            required: 'E-mail é obrigatório.'
          })}
        />
      </Box>
      <Box mb={2}>
        <TextField
          variant="standard"
          fullWidth={true}
          type="text"
          inputProps={{ 'data-testid': 'name' }}
          label="Nome"
          placeholder="Digite seu nome"
          error={!!hasError(errors.name)}
          helperText={hasError(errors.name)}
          {...register('name', {
            required: 'Nome é obrigatório.'
          })}
        />
      </Box>
      <Box mb={2}>
        <TextField
          type="password"
          variant="standard"
          inputProps={{ 'data-testid': 'password' }}
          fullWidth={true}
          autoComplete="off"
          label="Senha"
          placeholder="Digite sua senha"
          error={!!hasError(errors.password)}
          helperText={hasError(errors.password)}
          {...register('password', {
            minLength: {
              value: 9,
              message: 'Senha precisa ter no mínimo 9 caracteres.'
            },
            required: 'Senha é obrigatória.'
          })}
        />
      </Box>
      <Button
        disabled={loading}
        data-testid="submit"
        type="submit"
        variant="contained"
        color="secondary"
      >
        Cadastrar
      </Button>
    </form>
  );
}
