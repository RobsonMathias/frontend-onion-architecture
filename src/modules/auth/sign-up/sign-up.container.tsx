import React from 'react';

import { SignUpComponent } from './sign-up.component';
import { useSignUp } from '../hooks';
import { STORE_STATUS_LOADING } from 'constants/store';
import { SignUpPayload } from '@types';

export function SignUpContainer() {
  const { signUp, user } = useSignUp();

  function onSubmit(payload: SignUpPayload) {
    signUp(payload);
  }

  return <SignUpComponent onSubmit={onSubmit} loading={user.status === STORE_STATUS_LOADING} />;
}
