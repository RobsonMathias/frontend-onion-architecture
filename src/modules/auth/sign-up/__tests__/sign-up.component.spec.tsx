/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';

import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';
import '@fixtures/mocks';

import { setValueForm } from './fixture';
import { SignUpComponent } from '../sign-up.component';

describe('SignUpComponent', () => {
  it('should render successfully', () => {
    const onSubmit = jest.fn();
    const { container } = render(<SignUpComponent onSubmit={onSubmit} />);
    expect(container).toMatchSnapshot();
  });

  it('should submit form successfully', async () => {
    const onSubmit = jest.fn();
    const { getByTestId } = render(<SignUpComponent onSubmit={onSubmit} />);

    await act(async () => {
      setValueForm(fireEvent, getByTestId);
    });

    expect(onSubmit).toHaveBeenCalledWith({
      email: 'test@test.com',
      name: 'Test',
      password: '123456789'
    });
  });

  it('should render required errors', async () => {
    const onSubmit = jest.fn();
    const { getByTestId, container } = render(<SignUpComponent onSubmit={onSubmit} />);

    await act(async () => {
      fireEvent.click(getByTestId('submit'));
    });

    expect(container).toHaveTextContent('E-mail é obrigatório.');
    expect(container).toHaveTextContent('Senha é obrigatória.');
    expect(container).toHaveTextContent('Nome é obrigatório.');
  });
});
