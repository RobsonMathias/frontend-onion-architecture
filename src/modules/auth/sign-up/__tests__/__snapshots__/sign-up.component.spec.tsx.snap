// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpComponent should render successfully 1`] = `
<div>
  <form>
    <div
      class="MuiBox-root css-1qm1lh"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-1"
          id="mui-1-label"
        >
          E-mail
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            data-testid="email"
            id="mui-1"
            name="email"
            placeholder="email@exemplo.com"
            type="email"
            value=""
          />
        </div>
        
      </div>
    </div>
    <div
      class="MuiBox-root css-1qm1lh"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-2"
          id="mui-2-label"
        >
          Nome
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            data-testid="name"
            id="mui-2"
            name="name"
            placeholder="Digite seu nome"
            type="text"
            value=""
          />
        </div>
        
      </div>
    </div>
    <div
      class="MuiBox-root css-1qm1lh"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-3"
          id="mui-3-label"
        >
          Senha
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            data-testid="password"
            id="mui-3"
            name="password"
            placeholder="Digite sua senha"
            type="password"
            value=""
          />
        </div>
        
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
      data-testid="submit"
      tabindex="0"
      type="submit"
    >
      Cadastrar
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
